### DISCUSSION TITLE
File Nesting

### DISCUSSION THREAD

**Discussion by rawnly** [nvim-neo-tree/neo-tree.nvim#793] [Ideas]
I'm not into neovim plug-ins, and I'm not familiar with this code aspe, but shouldn't be difficult to support vscode like file nesting, maybe by adopting the same config 

https://code.visualstudio.com/updates/v1_67#_explorer-file-nesting

At the end of the day these are just "virtual" folders(?)

Also Xcode has the same feature by creating groups only in the ide but not as folders.

**Comment by cseickel** [comment]
We already have this feature. See `:h neo-tree-file-nesting`. For example:

```lua
    require("neo-tree").setup({
        nesting_rules = {
          ts = { ".d.ts", "js", "css", "html", "scss" }
        },
    })
```

**Reply by rawnly** [reply]
Thanks! Is there any docs other than the help? 

I'd like to group all my `*.config.js` under `package.json` but the API seems limited

**Reply by cseickel** [reply]
Sorry, it does not do that at all. It only nests files with a common base name.

I'm not even sure I would accept a PR where the nesting gets complex enough to do what you are looking for. I personally think it's an unnecessary level of complexity that would likely be a maintenance problem.

**Reply by rawnly** [reply]
Totally understandable, thanks ✌️

**Reply by mehalter** [reply]
@rawnly not sure if you are still interested in this, but a new PR has been merged in that actually enables this

```lua
    nesting_rules = {
      ["package.json"] = {
        pattern = "^package%.json$",
        files = { "*.config.js" },
      },
    },
```
